File: gluecodium/src/test/resources/smoke/generic_types/output/android/com/example/smoke/GenericTypesWithCompoundTypes.java
/*
 *

 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.List;
import java.util.Map;
import java.util.Set;
public final class GenericTypesWithCompoundTypes extends NativeBase {
    public enum SomeEnum {
        FOO(0),
        BAR(1);
        public final int value;
        SomeEnum(final int value) {
            this.value = value;
        }
    }
    public enum ExternalEnum {
        ON(0),
        OFF(1);
        public final int value;
        ExternalEnum(final int value) {
            this.value = value;
        }
    }
    public static final class BasicStruct {
        public double value;
        public BasicStruct(final double value) {
            this.value = value;
        }
    }
    public static final class ExternalStruct {
        @NonNull
        public String string;
        public ExternalStruct(@NonNull final String string) {
            this.string = string;
        }
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected GenericTypesWithCompoundTypes(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native List<GenericTypesWithCompoundTypes.ExternalStruct> methodWithStructList(@NonNull final List<GenericTypesWithCompoundTypes.BasicStruct> input);
    @NonNull
    public native Map<String, GenericTypesWithCompoundTypes.ExternalStruct> methodWithStructMap(@NonNull final Map<String, GenericTypesWithCompoundTypes.BasicStruct> input);
    @NonNull
    public native List<GenericTypesWithCompoundTypes.ExternalEnum> methodWithEnumList(@NonNull final List<GenericTypesWithCompoundTypes.SomeEnum> input);
    @NonNull
    public native Map<GenericTypesWithCompoundTypes.ExternalEnum, Boolean> methodWithEnumMapKey(@NonNull final Map<GenericTypesWithCompoundTypes.SomeEnum, Boolean> input);
    @NonNull
    public native Map<Integer, GenericTypesWithCompoundTypes.ExternalEnum> methodWithEnumMapValue(@NonNull final Map<Integer, GenericTypesWithCompoundTypes.SomeEnum> input);
    @NonNull
    public native Set<GenericTypesWithCompoundTypes.ExternalEnum> methodWithEnumSet(@NonNull final Set<GenericTypesWithCompoundTypes.SomeEnum> input);
    @NonNull
    public native List<DummyInterface> methodWithInstancesList(@NonNull final List<DummyClass> input);
    @NonNull
    public native Map<Integer, DummyInterface> methodWithInstancesMap(@NonNull final Map<Integer, DummyClass> input);
}

File: gluecodium/src/test/resources/smoke/generic_types/output/android-kotlin/com/example/smoke/GenericTypesWithCompoundTypes.kt
/*

 *
 */

@file:JvmName("GenericTypesWithCompoundTypesExtensions")


package com.example.smoke

import com.example.NativeBase

class GenericTypesWithCompoundTypes : NativeBase {

    enum class SomeEnum(@JvmField val value: Int) {
        FOO(0),
        BAR(1);
    }
    enum class ExternalEnum(@JvmField val value: Int) {
        ON(0),
        OFF(1);
    }
    class BasicStruct {
        @JvmField var value: Double



        constructor(value: Double) {
            this.value = value
        }





    }

    class ExternalStruct {
        @JvmField var string: String



        constructor(string: String) {
            this.string = string
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    external fun methodWithStructList(input: List<GenericTypesWithCompoundTypes.BasicStruct>) : List<GenericTypesWithCompoundTypes.ExternalStruct>


    external fun methodWithStructMap(input: Map<String, GenericTypesWithCompoundTypes.BasicStruct>) : Map<String, GenericTypesWithCompoundTypes.ExternalStruct>


    external fun methodWithEnumList(input: List<GenericTypesWithCompoundTypes.SomeEnum>) : List<GenericTypesWithCompoundTypes.ExternalEnum>


    external fun methodWithEnumMapKey(input: Map<GenericTypesWithCompoundTypes.SomeEnum, Boolean>) : Map<GenericTypesWithCompoundTypes.ExternalEnum, Boolean>


    external fun methodWithEnumMapValue(input: Map<Int, GenericTypesWithCompoundTypes.SomeEnum>) : Map<Int, GenericTypesWithCompoundTypes.ExternalEnum>


    external fun methodWithEnumSet(input: Set<GenericTypesWithCompoundTypes.SomeEnum>) : Set<GenericTypesWithCompoundTypes.ExternalEnum>


    external fun methodWithInstancesList(input: List<DummyClass>) : List<@JvmSuppressWildcards DummyInterface>


    external fun methodWithInstancesMap(input: Map<Int, DummyClass>) : Map<Int, @JvmSuppressWildcards DummyInterface>




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


